extensions/clearinghouse/src/main/java/de/sovity/extension/clearinghouse/sender/LogMessageSender.java

```diff
@@ -49,15 +51,13 @@ public Message buildMessageHeader(LogMessage logMessage,
 
     @Override
     public String buildMessagePayload(LogMessage logMessage) throws Exception {
-        var jo = new JSONObject();
-        jo.put("AgreementId", logMessage.contractAgreement().getId());
-        jo.put("AssetId", logMessage.contractAgreement().getAssetId());
-        jo.put("ContractStartDate", logMessage.contractAgreement().getContractStartDate());
-        jo.put("ContractEndDate", logMessage.contractAgreement().getContractEndDate());
-        jo.put("ContractSigningDate", logMessage.contractAgreement().getContractSigningDate());
-        jo.put("ConsumerAgentId", logMessage.contractAgreement().getConsumerAgentId());
-        jo.put("ProviderAgentId", logMessage.contractAgreement().getProviderAgentId());
-        return jo.toString();
+        if (logMessage.eventToLog() instanceof ContractAgreement contractAgreement) {
+            return buildContractAgreementPayload(contractAgreement);
+        } else if (logMessage.eventToLog() instanceof TransferProcess transferProcess) {
+            return buildTransferProcessPayload(transferProcess);
+        } else {
+            return null;
```

Comment:
Throw an EDCException with a message like "ObjectType {class-name} not support in LogMessageSender"